outputs/kubeless.go

```diff
@@ -0,0 +1,89 @@
+package outputs
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"log"
+	"strconv"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/falcosecurity/falcosidekick/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// NewKubelessClient returns a new output.Client for accessing Kubernetes.
+func NewKubelessClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
+	if config.Kubeless.Kubeconfig != "" {
+		restConfig, err := clientcmd.BuildConfigFromFlags("", config.Kubeless.Kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+		clientset, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return nil, err
+		}
+		return &Client{
+			OutputType:       "Kubeless",
+			Config:           config,
+			Stats:            stats,
+			PromStats:        promStats,
+			StatsdClient:     statsdClient,
+			DogstatsdClient:  dogstatsdClient,
+			KubernetesClient: clientset,
+		}, nil
+	}
+	return NewClient(
+		"Kubeless",
+		"http://"+config.Kubeless.Function+"."+config.Kubeless.Namespace+".svc.cluster.local:"+strconv.Itoa(config.Kubeless.Port),
+		config,
+		stats,
+		promStats,
+		statsdClient,
+		dogstatsdClient,
+	)
+}
+
+// KubelessCall .
+func (c *Client) KubelessCall(falcopayload types.FalcoPayload) {
+	c.Stats.Kubeless.Add(Total, 1)
+
+	if c.Config.Kubeless.Kubeconfig != "" {
+		str, _ := json.Marshal(falcopayload)
+		req := c.KubernetesClient.CoreV1().RESTClient().Post().AbsPath("/api/v1/namespaces/" + c.Config.Kubeless.Namespace + "/services/" + c.Config.Kubeless.Function + ":" + strconv.Itoa(c.Config.Kubeless.Port) + "/proxy/").Body(str)
+		b := make([]byte, 11)
+		rand.Read(b)
+		base64.RawURLEncoding.EncodeToString(b)
+		req.SetHeader("event-id", base64.RawURLEncoding.EncodeToString(b))
```

Comment:
Is there a reason to do it this way and not with a [uuid library](https://github.com/google/uuid)?

Reply:
Only because I didn't know that package. Feel to propose a suggestion 😉

Reply:
> lgtm, just some nits:
> 
>     * squash the commits to have just one commit
> 
>     * and just a comment, try to avoid to update dependencies in the PR that does other thing, if the PR need a new dependency is fine, but avoid update non-releated dependencies, we can open a separated PR for that

I squashed the commits, you're right.

For the dependencies, I faced big conflicts while I was developping (in fact, I extracted some snippets from `kubeless cli` before rewriting from scratch everything), and that's a relicate. You're right 😉 . Sorry.